demo.py

```diff
@@ -13,213 +13,103 @@
-    print('Tree depth:', TREE_DEPTH)
-
-
-    # input: dataset X and labels y (in {+1, -1}
-
-    # Preprocessing data
-    # imputer = SimpleImputer()
-    scaler = preprocessing.StandardScaler()     # standardize features
-    normalizer = preprocessing.Normalizer()     # normalize samples
-
-    # X = imputer.fit_transform(X)
-    X_train = scaler.fit_transform(X_train)
-    X_train = normalizer.fit_transform(X_train)
-
-    # X_test = imputer.fit_transform(X_test)
-    X_test = scaler.fit_transform(X_test)
-    X_test = normalizer.fit_transform(X_test)
-
-
-    # ===============================================
-    print('\nAdaboost:')
-
-    clf = AdaBoostClassifier(n_estimators=NUM_WEAK_CLASSIFIERS)
+from qboost import QBoostClassifier, qboost_lambda_sweep
+from datasets import make_blob_data, get_handwritten_digits_data
 
-    # scores = cross_val_score(clf, X, y, cv=5, scoring='accuracy')
-    clf.fit(X_train, y_train)
 
-    hypotheses_ada = clf.estimators_
-    # clf.estimator_weights_ = np.random.uniform(0,1,size=NUM_WEAK_CLASSIFIERS)
-    y_train_pred = clf.predict(X_train)
-    y_test_pred = clf.predict(X_test)
-
-    print_accuracy(y_train, y_train_pred, y_test, y_test_pred)
-
-
-    # ===============================================
-    print('\nDecision tree:')
-
-    clf2 = WeakClassifiers(n_estimators=NUM_WEAK_CLASSIFIERS, max_depth=TREE_DEPTH)
-    clf2.fit(X_train, y_train)
-
-    y_train_pred2 = clf2.predict(X_train)
-    y_test_pred2 = clf2.predict(X_test)
-
-    if verbose:
-        print('weights:\n', clf2.estimator_weights)
-
-    print_accuracy(y_train, y_train_pred2, y_test, y_test_pred2)
-
-
-    # ===============================================
-    print('\nQBoost:')
-
-    DW_PARAMS = {'num_reads': NUM_READS,
-                 'auto_scale': True,
-                 # "answer_mode": "histogram",
-                 'num_spin_reversal_transforms': 10,
-                 # 'annealing_time': 10,
-                 }
-
-    clf3 = QBoostClassifier(n_estimators=NUM_WEAK_CLASSIFIERS, max_depth=TREE_DEPTH)
-    clf3.fit(X_train, y_train, emb_sampler, lmd=lmd, **DW_PARAMS)
-
-    y_train_dw = clf3.predict(X_train)
-    y_test_dw = clf3.predict(X_test)
-
-    if verbose:
-        print('weights\n', clf3.estimator_weights)
-
-    print_accuracy(y_train, y_train_dw, y_test, y_test_dw)
-
-
-    # ===============================================
-    print('\nQBoostPlus:')
-    clf4 = QBoostPlus([clf, clf2, clf3])
-    clf4.fit(X_train, y_train, emb_sampler, lmd=lmd, **DW_PARAMS)
-    y_train4 = clf4.predict(X_train)
-    y_test4 = clf4.predict(X_test)
+if __name__ == '__main__':
+    import argparse
 
-    if verbose:
-        print('weights\n', clf4.estimator_weights)
+    parser = argparse.ArgumentParser(description="Run QBoost example", epilog="Information about additional options that are specific to the data set can be obtained using either 'demo.py blobs -h' or 'demo.py digits -h'.")
```

Comment:
Can this line with almost 200 chars  (and a few other long lines below) be broken across a few lines?

Reply:
Absolutely.  I went ahead and ran demo.py and qboost.py through a formatter to clean up long lines.  It didn't split up strings though, so we still have things like:
https://github.com/mcfarljm/qboost/blob/447cee72545380be27de50c037d9fc8dd2d631ca/demo.py#L32-L33